### ISSUE TITLE
Incompatibility with Menhir 20180530

### ISSUE THREAD

**Issue by ildyria** [AbsInt/CompCert#234]
# Proof of Bug

The following:

```bash
cd ~/
opam install menhir
git clone https://github.com/AbsInt/CompCert/
cd CompCert
./configure -clightgen -coqdevdir ~/.opam/Tweetnacl/lib/coq/user-contrib/compcert ia32-linux -bindir ~/.opam/Tweetnacl/bin -libdir ~/.opam/Tweetnacl/lib/compcert -ignore-coq-version
```

Leads to:

```bash
Testing assembler support for CFI directives... yes
Testing linker support for '-no-pie' / '-nopie' option... no
Testing Coq... version 8.8.0 -- good!
Testing OCaml... version 4.06.0 -- good!
Testing OCaml .opt compilers... yes
Testing Menhir... version 20180530 -- good!
Testing GNU make... version 4.1 (command 'make') -- good!

CompCert configuration:
    Target architecture........... x86
    Hardware model................ 32sse2
    Application binary interface.. standard
    Endianness.................... little
    OS and development env........ linux
    C compiler.................... gcc -m32
    C preprocessor................ gcc
    Assembler..................... gcc
    Assembler supports CFI........ true
    Assembler for runtime lib..... gcc -m32 -c
    Linker........................ gcc
    Linker needs '-no-pie'........ false
    Math library.................. -lm
    Build command to use.......... make
    Binaries installed in......... /home/biv/.opam/Tweetnacl/bin
    Runtime library provided...... true
    Library files installed in.... /home/biv/.opam/Tweetnacl/lib/compcert
    Standard headers provided..... true
    Standard headers installed in. /home/biv/.opam/Tweetnacl/lib/compcert/include
    Coq development installed in.. /home/biv/.opam/Tweetnacl/lib/coq/user-contrib/compcert
```

Then we can do a `make -j`:

```bash
make -j   
make[1]: Entering directory '/home/biv/CompCert'
ocamlopt -o tools/ndfun str.cmxa tools/ndfun.ml
menhir --coq cparser/Parser.vy
Preprocessing x86/ConstpropOp.vp
Preprocessing x86/SelectOp.vp
Preprocessing x86/SelectLong.vp
Preprocessing backend/SelectDiv.vp
Preprocessing backend/SplitLong.vp
Analyzing Coq dependencies
*** Warning: in file cparser/Parser.v, library MenhirLib.Tuples is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Alphabet is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Grammar is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Automaton is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Main is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Tuples is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Alphabet is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Grammar is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Automaton is required and has not been found in the loadpath!
*** Warning: in file cparser/Parser.v, library MenhirLib.Main is required and has not been found in the loadpath!
make[1]: Leaving directory '/home/biv/CompCert'
make proof
make[1]: Entering directory '/home/biv/CompCert'
COQC lib/Axioms.v
COQC lib/Coqlib.v
COQC flocq/Core/Fcore_Zaux.v
COQC driver/Compopts.v
COQC cparser/validator/Alphabet.v
COQC cparser/validator/Tuples.v
COQC cparser/Cabs.v
COQC lib/Wfsimpl.v
COQC cparser/Parser.v
File "./cparser/Parser.v", line 11, characters 30-36:
Error:
Cannot find a physical path bound to logical path matching suffix
<> and prefix MenhirLib.

Makefile:195: recipe for target 'cparser/Parser.vo' failed
make[1]: *** [cparser/Parser.vo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/biv/CompCert'
Makefile:136: recipe for target 'all' failed
make: *** [all] Error 2
```

# Downgrading Menhir

However this works:

```bash
opam install menhir.20180528
make clean
# reconfigure
./configure -clightgen -coqdevdir ~/.opam/Tweetnacl/lib/coq/user-contrib/compcert ia32-linux -bindir ~/.opam/Tweetnacl/bin -libdir ~/.opam/Tweetnacl/lib/compcert -ignore-coq-version
make -j
```

**Comment by ildyria** [comment]
related: https://gitlab.inria.fr/fpottier/menhir/blob/master/CHANGES.md

**Comment by xavierleroy** [comment]
@jhjourdan , this one is your responsibility.

**Comment by ejgallego** [comment]
This is also breaking Coq's CI.

**Comment by xavierleroy** [comment]
Fixed by #235 and #236.  The master branch of CompCert is now compatible with the latest Menhir, 20180530, but also with previous versions starting with 20161201.  Thanks @ildyria and @jhjourdan for your help.
